### PR TITLE
Order notes meta box: clarify visibility, modernize bubble styling

### PR DESCRIPTION
**Pull request by j111q**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

The "Note to customer" action on the order edit screen emails the customer immediately, but nothing in the UI signals that consequence until after the click. [Real-world feedback](https://x.com/PineDigitalCo/status/1910348409426358379) flagged how easy it is to misuse — a private internal note typed in haste gets sent to the customer.

This PR addresses the visibility clarity and modernizes the surrounding UI.

#### Form changes

- Visible **"Visibility"** label above the type dropdown (was screen-reader-only)
- Dynamic button label that reflects what will actually happen: **"Add private note"** / **"Send note to customer →"**
- Full-width button (secondary, not primary). The button-primary class is deliberately omitted — a primary button felt too aggressive for a destructive-by-default action like "Send note to customer". Full-width is less common in classic wp-admin meta boxes, but there's solid precedent for full-width actions in narrow sidebar contexts — the block editor's post-publish panel and the Customizer's Save & Publish button both use it. Visual hierarchy reads cleaner at full width, and the prominence suits an action with an irreversible side effect (sending the customer email).
- **"Preview or edit email template ↗"** helper link, only shown when "Public note to customer" is selected (opens in a new tab) — gives merchants quick access to customize the email that's about to go out
- Dropdown option re-labeled "Note to customer" → **"Public note to customer"** for clearer scope
- **Removed the help-tooltip** that was next to the "Add note" label. Its content ("Add a note for your reference, or add a customer note (the user will be notified)") is now carried more clearly by the visible Visibility label, the dynamic CTA, and the email template helper link.

#### Note rendering changes

- Customer notes get a **"Sent to customer"** header band inside the bubble with a supplementary yellow tint (matches `@wordpress/components` Notice `is-warning`), making it impossible to miss that this got emailed
- Bubble restyle: thin gray border (`$gray-200`), no chat-bubble tail. The bubble bg now communicates **how the note got here**, not just what kind it is:
  - **System notes** (auto-generated by WC: status changes, payment events, webhook callbacks) sit on a muted gray bg (`$gray-100`) — visually de-emphasized so manual notes scan more easily on busy orders.
  - **Manual private notes** (a human deliberately typed) sit on white — primary content weight.
  - **Customer notes** (manual + sent via email) sit on white with a yellow header band — same content weight as private, with the "this left the building" marker.
- This corrects an old WC palette where system notes were given a dusty pink `#d7cad2` background — semantically misleading (red-adjacent for routine auto-events).
- Meta line ("Apr 17, 2025 by admin Delete") rendered in 11px muted gray, hits AA contrast at 4.5:1
- Bug fix: empty state ("There are no notes yet") now reappears after deleting the last note (was a real bug — the JS removed the `<li>` without restoring the empty state)

#### Confirmation copy

- Customer-note delete confirmation now warns that the email cannot be recalled:
  > Are you sure you wish to delete this note? This action cannot be undone. Caution: This only removes the note from your records — it does not recall the email already sent to the customer.
- Private-note confirmation unchanged

#### Accessibility improvements

- **Visibility label is now visible to sighted users.** The original markup used `class="screen-reader-text"` on the dropdown's label, so anyone not using a screen reader had no visual indication of what the dropdown controlled. Now visible to everyone.
- **Customer-note differentiation no longer relies on color alone.** Previously a customer note was distinguished from a private note only by the bubble's blue background — a [WCAG 1.4.1](https://www.w3.org/WAI/WCAG21/Understanding/use-of-color.html) failure for users who don't perceive color (colorblindness, grayscale displays, high-contrast modes). Now the differentiation is carried primarily by a "Sent to customer" text label in a header band above the note body. The supplementary yellow tint (`#fef8ee` — mirrors `@wordpress/components` Notice `is-warning` variant) reinforces the cue visually but isn't load-bearing — text labels remain visible on grayscale/high-contrast displays.
- **Meta line text now passes WCAG AA contrast.** Original was `color: #999` on the white meta-box background — **2.85:1**, fails AA (which requires 4.5:1 for normal text). New color `$subtext` (`#767676`) hits **4.51:1** on white.

#### Known inconsistency (deliberate, follow-up in flight)

The form labels in this meta box ("Add note", "Visibility") don't visually match labels elsewhere on the order edit page (e.g. "Date created", which still uses the older bold-label pattern). This is intentional — labels here align with the WordPress 7.0 admin design language. The follow-up to bring the rest of the order edit page into the same style is open as [#64564](https://github.com/woocommerce/woocommerce/pull/64564) (Align Order Edit meta boxes with WordPress 7.0 design conventions). Once both land, the page reads consistently end-to-end.

#### Tracks events

Adds a new `wcadmin_order_edit_delete_order_note` event in the JS delete handler. Mirrors the existing `wcadmin_order_edit_add_order_note` event (same `order_id`, `note_type`, `status` props) with one extra prop:

- `seconds_since_added` — seconds elapsed between the note's creation timestamp and the delete action. Approximate (browser-local clock vs server-rendered timestamp), null on parse failure. Sufficient for a coarse "deleted shortly after creation" derived metric.

The intent is to enable a **"regret rate"** derived metric — customer notes deleted within a short window of being added are likely accidental sends. The email itself can't be recalled, but the visible record can be scrubbed; tracking this gives us a rough proxy for how often the original UI clarity issue was actually causing harm. Pre/post comparison of this metric should let us measure whether the redesign reduced misuse.

Registration: a parallel PR is being opened to `Automattic/tracks-events-registration` to register the event metadata in the Tracks Event Directory. Linked from the registration PR back to this one via the `diff` field.

#### i18n note

The button arrow `→` is included inside the translated string `__('Send note to customer →')` rather than concatenated outside. This lets translators preserve, replace (e.g. `←` for RTL), or drop the arrow as appropriate per locale.

#### Tokenization

- Adds `$gray-100` / `$gray-200` to legacy `_variables.scss` (mirrors `@wordpress/base-styles/_colors.scss`)
- Overlaps benignly with [#64280](https://github.com/woocommerce/woocommerce/pull/64280) (which adds `$gray-700` / `$gray-900` — disjoint sets, whoever merges first wins on `_variables.scss`, no visual conflict either way)

#### Code health

- The AJAX handler that renders newly-added notes had a copy of the LI markup that had drifted from the template. Both paths now produce equivalent markup. They remain two separate copies — full deduplication is out of scope for this PR; flagging as a follow-up.

#### Notes for extension authors

This PR changes the structural CSS of order notes. No PHP hooks, action signatures, REST endpoints, or class names are removed — the changes are markup/styling only. Extensions targeting the following CSS will see cosmetic effects:

- **`.note_content::after` removed.** The chat-bubble tail pseudo-element is gone. Plugins like `woocommerce-colored-order-notes` that color the tail will silently no-op for that selector. Body coloring via `.note_content` background still works.
- **Dotted underline on `.exact-date` removed.** Plugins targeting `.exact-date { border-bottom }` will no-op.
- **New `.note_header`, `.note_body`, `.note_label` wrappers inside `.note_content`.** Extensions selecting `.note_content > p` directly will still match (the body content is inside `.note_body`, but the inline `<p>` produced by `wpautop()` is unchanged). Extensions wanting to reach the bubble's text content directly should prefer `.note_body p`.
- **System-note background changed.** Was `#d7cad2` (pinkish/mauve), now `$gray-100` (light gray) — system notes are still visually distinct from private (white) and customer (white + yellow band) notes, just with a different shade. Plugins overriding `li.system-note .note_content { background }` will still apply.

None of these are hard breaks — markup contracts and class names are preserved.

#### Constraints

- No Gutenberg component imports — keeps the order edit page light. Visuals achieved with classic wp-admin elements + Dashicons.
- All Dashicons + colors used are available on every WP version WC currently supports — no version-gating needed. Tested on WP 6.9 and WP 7.0+ — the meta box itself renders consistently in both, sitting naturally inside whatever surrounding admin chrome the WP version provides (postbox borders, default font, button heights inherit from core).

### Screenshots or screen recordings:

image

image

### How to test the changes in this Pull Request:

1. Open any order in wp-admin → **Orders → [order] → Order notes** (right sidebar meta box)
2. **Form behavior:**
   - Confirm "Visibility" label is visible above the dropdown
   - Switch dropdown to "Public note to customer" — button changes to "Send note to customer →", and a "Preview or edit email template ↗" helper link appears under the dropdown (opens in a new tab when clicked)
   - Switch back to "Private note" — button reverts to "Add private note", helper link disappears
3. **Add a private note** — appears as a gray bubble (no header), with meta line ("Apr 17, 2025 at 9:36 am by admin Delete") below in muted gray
4. **Add a public note** — appears with a "Sent to customer" header band (yellow tint), a thin divider, then the body
5. **Delete a customer note** — confirmation dialog includes the "Caution: ... cannot recall email already sent" warning; private notes get the original short confirmation
6. **Delete the last note** — empty state ("There are no notes yet") should reappear immediately
7. **HPOS:** repeat all of the above on the HPOS edit screen (`?page=wc-orders&action=edit&id=…`) — same meta box, should look identical
8. **Cross-version:** the meta box reads correctly on WP 6.9 and WP 7.0+, sitting naturally inside each version's admin chrome

### Testing that has already taken place:

- [x] Local testing on WP 7.x (HPOS-enabled)
- [x] Local testing on WP 6.9
- [x] `pnpm lint:changes:branch` clean (0 errors; pre-existing warnings unrelated)
- [x] `phpstan analyse` clean on all modified PHP files
- [x] `composer exec phpcs-changed` passing
- [x] e2e spec `order-edit.spec.ts` updated to match new button labels

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

### Changelog entry

- [x] Added: `plugins/woocommerce/changelog/64999-sprinkle-order-notes-visibility`

### CODE REVIEW

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js [hunk 1]
```diff
@@ -1359,8 +1359,24 @@ jQuery( function ( $ ) {
...
 				.on( 'click', 'button.add_note', this.add_order_note )
-				.on( 'click', 'a.delete_note', this.delete_order_note );
+				.on( 'click', 'a.delete_note', this.delete_order_note )
+				.on( 'change', 'select#order_note_type', this.update_note_type_ui );
 
+			// Sync the CTA + helper link to reflect the current select state on load.
+			$( 'select#order_note_type' ).trigger( 'change' );
+		},
+
+		update_note_type_ui: function() {
```
Comment (Aljullu) [inline plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js:1369]: Not a blocker, but for the future: it's better to use vanilla JS instead of jQuery if possible. Nowadays there is little benefit on using jQuery over vanilla JS, and vanilla JS is usually more performant and makes it easy to reuse the code in the future.

File: plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts [hunk 2]
```diff
@@ -220,7 +220,7 @@ test.describe( 'Edit order', { tag: [ tags.SERVICES, tags.HPOS ] }, () => {
 
 		// delete the note
 		await page
-			.getByRole( 'button', { name: 'Delete note' } )
+			.getByRole( 'button', { name: /^Delete (system )?note/ } )
```
Comment (Aljullu) [inline plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts]: This is testing merchant-added notes, so I think it's good that we verify the button label is `Delete note` instead of `Delete system note`.

```suggestion
			.getByRole( 'button', { name: 'Delete note' } )
```

File: plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts [hunk 3]
```diff
@@ -250,14 +254,14 @@ test.describe( 'Edit order', { tag: [ tags.SERVICES, tags.HPOS ] }, () => {
...
```
Comment (Aljullu) [inline plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts]: Same here: 🙂 

```suggestion
			.getByRole( 'button', { name: 'Delete note' } )
```

File: plugins/woocommerce/includes/class-wc-ajax.php [hunk 4]
```diff
@@ -1705,22 +1705,31 @@ public static function add_order_note() {
...
+						?>
+					</div>
 				</div>
 				<p class="meta">
-					<abbr class="exact-date" title="<?php echo esc_attr( $note->date_created->date( 'y-m-d h:i:s' ) ); ?>">
+					<abbr class="exact-date" title="<?php echo esc_attr( $note->date_created->date( 'Y-m-d H:i:s' ) ); ?>">
 						<?php
-						/* translators: $1: Date created, $2 Time created */
-						printf( esc_html__( 'added on %1$s at %2$s', 'woocommerce' ), esc_html( $note->date_created->date_i18n( wc_date_format() ) ), esc_html( $note->date_created->date_i18n( wc_time_format() ) ) );
+						/* translators: %1$s: note date %2$s: note time */
```
Comment (Aljullu) [inline plugins/woocommerce/includes/class-wc-ajax.php]: I know the original `translators` text wasn't very explicit. 😅 But I think that mentioning that these are order notes would make localization easier:

```suggestion
						/* translators: %1$s: order note date %2$s: order note time */
```

**Review by Aljullu — APPROVED** [review]
Really nice improvements! 👏  I tested it, and everything works great on my end. The code looks good as well. I left a few minor comments, but pre-approving.

Image

File: plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js (hunk 1 above)
Comment (j111q) [reply plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js:1369]: thank you so much for the tip and for reviewing, albert! saved in memory for future PRs 😄

File: plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts (hunk 2 above)
Comment (j111q) [reply plugins/woocommerce/tests/e2e-pw/tests/order/order-edit.spec.ts]: Done in [96995b9b](https://github.com/woocommerce/woocommerce/commit/96995b9b81). Thank you!!

**Comment by PV-Web** [comment]
Hi, I like the changes but a few issues for my use.

First we use the note on the main meta box (not de smaller side box). This stretches the dropdown box and button too much

image

Next I also have other employee use this section and I dont want them to have a link to the settings to the email template.


Also it would be nice if there was more contrast between the private notes and the notes being made by the system itself.

See this screenshot:
image

So I would suggest to remove the new feature (link to template) and make the max width of the buttons the width of the side meta box. and add a different contrast color for private notes.

open for suggestion and your insights.

**Comment by beaulebens** [comment]
Thanks for the feedback @PV-Web! On the email template link; what type of WordPress/WooCommerce user is your other employee? We should at least add in some control here so that you only see that link if you have permission to edit it (e.g. a Store Manager or Site Admin); would that fix things for your case?

And good catch on the width of the button/drop-down; we can probably do something like make them max out at 50% wide each, and stack side by side or something.

**Comment by PV-Web** [comment]
We have a sales/store manager but the adman has already setup everything regarding mail/templates etc. So to prevent any mistakes I would rather leave it out. Also I would leave it out for the admin, as he already has a way to get there via the settings. Keeps the backend clean. Or only show it to the admin when there is no custom text yet added to the template fields. (this means it has not been set yet and it is a new install)



Also, and I have thinking about this for a long time already, would it be possible to add templates for repeating messages.
Like in the same drop-down box a new options like: "Shipping address unclear" which would auto fill in the message ( not send it directly: 

"Thank you for your order. We have checked your address but our shipping system unfortunately cannot find it etc etc."

or like a default template like (dropdown option "default reply")
"Let me know if you have any questions.
You can simply reply to this email and I'm happy to help you.
Regards, "

I will prefill the message but not send it yet. Also the setting if it will be send out to a customer or internal should be set by the template.

If this would be something only like uploading a template file to the server to add it etc. does not have to be fancy and no UI needed.
I think many admins/sales managers would greatly appreciate it to keep the internals and customer notes consistent.

We now do this manually by copying text from private page on our website and pasting it into the message field. We have 12 default messages.

What would your thoughts be on this?
Regards.

**Comment by beaulebens** [comment]
> only show it to the admin when there is no custom text yet added to the template fields. (this means it has not been set yet and it is a new install)

This one probably makes sense.

The request for templates for repeating messages sounds very much like a separate issue, so I'd encourage you to please take a look at the feature requests board, and add it there if relevant: https://woocommerce.com/feature-requests/woocommerce/
